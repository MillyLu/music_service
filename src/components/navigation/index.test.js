/* eslint-disable no-undef */
import React from "react";
import { render} from "@testing-library/react";

import Navigation from "./nav";


describe("<Navigation />", () => {
    it("should render successfully and match snapshot", () => {

      // Cache original functionality
const realUseState = React.useState

// Stub the initial state
const stubInitialState = ['isMenuOpen']

// Mock useState before rendering your component
jest
  .spyOn(React, 'useState')
  .mockImplementationOnce(() => realUseState(stubInitialState));

      const { container } = render(<Navigation></Navigation>);
  
      expect(container).toMatchSnapshot();


    });
  });