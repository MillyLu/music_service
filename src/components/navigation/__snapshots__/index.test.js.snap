// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should render successfully and match snapshot 1`] = `
<div>
  <nav
    class="sc-bcXHqe eBRLil"
    style="background-color: rgb(28, 28, 28);"
  >
    <div
      class="sc-jSUZER iEkKUm"
    >
      <img
        alt="logo"
        class="sc-gKPRtg cNQYAP"
        data-testid="logo"
        src="logoLight.png"
      />
    </div>
    <div
      aria-hidden="true"
      class="sc-dkrFOg jFouIx"
      role="button"
      tabindex="0"
    >
      <span
        class="sc-hLBbgP jlueLK"
        style="background-color: rgb(255, 255, 255);"
      />
      <span
        class="sc-hLBbgP jlueLK"
        style="background-color: rgb(255, 255, 255);"
      />
      <span
        class="sc-hLBbgP jlueLK"
        style="background-color: rgb(255, 255, 255);"
      />
    </div>
  </nav>
</div>
`;
